lib/graphql.mli

```diff
@@ -1,15 +1,18 @@
-type request = {
-  meth : Curly.Meth.t;
-  url : string;
-  headers : Curly.Header.t;
-  body : Yojson.Safe.t;
-}
+module Request : sig
+  type t
 
-val request :
-  ?variables:(string * Yojson.Safe.t) list ->
-  token:string ->
-  query:string ->
-  unit ->
-  request
+  val make :
+    ?variables:(string * Yojson.Safe.t) list ->
+    token:string ->
+    query:string ->
+    unit ->
+    t
 
-val exec : request -> (Yojson.Safe.t, [ `Msg of string ]) result
+  val exec :
+    Cohttp_eio.Client.t ->
+    Eio.Switch.t ->
```

Comment:
You shouldn't need a switch here; just create one inside the function (see https://github.com/ocaml-multicore/eio?tab=readme-ov-file#switches-1).

Reply:
Thanks, I must have missed this bit in the doc!